mongo/mongo.go

```diff
@@ -124,6 +125,47 @@ func (m *Mongo) AddReview(review *models.Review) {
 
 	return
 }
+// UpdateReview updates an existing Review.
+// Only the message and user can be updated.
+// It returns an error if the review is not found
+func (m *Mongo) UpdateReview(ctx context.Context, reviewID string, review *models.Review) error {
+	s := m.Session.Copy()
+	defer s.Close()
+
+	logData := log.Data{
+		"review_id":  reviewID,
+		"database":   m.Database,
+		"collection": m.ReviewsCollection}
+
+	updates := make(bson.M)
+	if review.Message != "" {
+		updates["message"] = review.Message
+	}
+
+	if review.User != (models.User{}) {
+		if review.User.Forenames != "" {
+			updates["user.forenames"] = review.User.Forenames
+		}
+		if review.User.Surname != "" {
+			updates["user.surname"] = review.User.Surname
+		}
+	}
+
+	if len(updates) > 0 {
+		updates["last_updated"] = time.Now().UTC()
+	}
+
+	update := bson.M{"$set": updates}
+	if err := s.DB(m.Database).C(m.ReviewsCollection).UpdateId(reviewID, update); err != nil {
+		if err == mgo.ErrNotFound {
+			log.Event(ctx, apierrors.ErrReviewNotFound.Error(), log.ERROR, log.Error(err), logData)
+			return apierrors.ErrReviewNotFound
```

Comment:
Here you are using the `apierrors` package which I'd advise against. This creates a dependency between the `api` and the `datastore` package and tightly couples them together. These packages should be independent of one another.

The datastore and api are 2 different things and have different concerns. So the datastore has no business knowing what the API will do with a not found error, its not its concern. It should return the appropriate error to the caller and then its up to the caller how they handle it. 
This may sound like a nit picky point but it is a good practice principle to follow and it will help you avoid design issues and cyclic dependencies.

My advice would be to create a `datastore.ErrNotFound` var and return that instead. This decouples the `datastore` from the `api` and also lets the API about the specific case where a document is not found (addressing one of my previous points).